### PR TITLE
feat(Homebrew): AND-1435 Get maximum spendable with only an account reference

### PR DESCRIPTION
**Pull request by westonal**

### CODE REVIEW

File: core/src/main/java/com/blockchain/datamanagers/TransactionSendDataManager.kt [hunk 1]
```diff
@@ -59,22 +61,26 @@ class TransactionSendDataManager(
...
-            feeType
-        )
-        CryptoCurrency.BCH -> getMaxBitcoinCash(
-            account as GenericMetadataAccount,
-            fees as BitcoinLikeFees,
-            feeType
-        )
-        CryptoCurrency.ETHER -> getMaxEther(fees as EthereumFees)
+    ) = getMaximumSpendable(
+        account.toAccountReference(),
```
Comment (westonal) [inline core/src/main/java/com/blockchain/datamanagers/TransactionSendDataManager.kt]: so trivial, could depreciate

File: core/src/main/java/com/blockchain/datamanagers/TransactionSendDataManager.kt [hunk 2]
```diff
@@ -59,22 +61,26 @@ class TransactionSendDataManager(
     }
 
     fun getMaximumSpendable(
-        cryptoCurrency: CryptoCurrency,
```
Comment (westonal) [inline core/src/main/java/com/blockchain/datamanagers/TransactionSendDataManager.kt]: Finding currency from account is possible.

File: core/src/main/java/com/blockchain/datamanagers/TransactionSendDataManager.kt [hunk 3]
```diff
@@ -143,31 +149,19 @@ class TransactionSendDataManager(
...
-        .doOnError { Timber.e(it) }
-        .onErrorReturn { CryptoValue.ZeroBch }
-
-    private fun getMaxBchOrBtc(
-        xPub: String,
-        cryptoCurrency: CryptoCurrency,
-        fees: BitcoinLikeFees,
-        feeType: FeeType
-    ): Single<BigInteger> = getUnspentOutputs(xPub, cryptoCurrency)
-        .map { sendDataManager.getMaximumAvailable(it, fees.feeForType(feeType)).left }
```
Comment (westonal) [inline core/src/main/java/com/blockchain/datamanagers/TransactionSendDataManager.kt]: These three functions are now one.

File: wallet/src/main/java/info/blockchain/wallet/payload/data/Account.java [hunk 4]
```diff
@@ -22,7 +22,7 @@
 public class Account implements JsonSerializableAccount {
 
     @JsonProperty("label")
-    private String label;
+    private String label = "";
```
Comment (westonal) [inline wallet/src/main/java/info/blockchain/wallet/payload/data/Account.java]: The tests were failing with null labels dues to converting to account references, and I didn't want to go through them all. But will do if you want to keep as null.

File: core/src/main/java/com/blockchain/datamanagers/TransactionSendDataManager.kt [hunk 5]
```diff
@@ -33,57 +37,69 @@ class TransactionSendDataManager(
...
             account as GenericMetadataAccount,
-            fees.feeForType(feeType)
+            (fees as BitcoinLikeFees).feeForType(feeType)
         )
     }
 
     fun getMaximumSpendable(
-        cryptoCurrency: CryptoCurrency,
         account: JsonSerializableAccount,
-        fees: FeeOptions,
```
Comment (ditn) [inline core/src/main/java/com/blockchain/datamanagers/TransactionSendDataManager.kt]: Ah woops I missed this one, thanks.

File: morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/confirmation/ExchangeConfirmationFragment.kt [hunk 6]
```diff
@@ -95,6 +100,7 @@ class ExchangeConfirmationFragment :
         compositeDisposable += exchangeModel
             .exchangeViewModels
             .observeOn(AndroidSchedulers.mainThread())
+            .doOnNext { it.from }
```
Comment (ditn) [inline morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/confirmation/ExchangeConfirmationFragment.kt]: What's this?

File: wallet/src/main/java/info/blockchain/wallet/payload/data/Account.java (hunk 4 above)
Comment (ditn) [reply wallet/src/main/java/info/blockchain/wallet/payload/data/Account.java]: Nah this is fine with me I think.

File: morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/confirmation/ExchangeConfirmationFragment.kt (hunk 6 above)
Comment (westonal) [reply morph/ui/src/main/java/com/blockchain/morph/ui/homebrew/exchange/confirmation/ExchangeConfirmationFragment.kt]: good question

**Pull request merged**